gapis/api/vulkan/resources.go

```diff
@@ -1080,8 +1084,12 @@ func commonShaderDataGroups(ctx context.Context,
 			module := stage.Module()
 
 			words, _ := module.Words().Read(ctx, nil, s, nil)
-			source := shadertools.DisassembleSpirvBinary(words)
-			shader := &api.Shader{Type: api.ShaderType_Spirv, Source: source}
+			source, sourceLanguage, isCross, err := shadertools.ExtractDebugSource(words)
+			if err != nil {
+				source = err.Error()
```

Comment:
Don't return the error as the source. Just return an empty string and handle it in the UI to not show the source tab.
You can log the error, so it's not lost, and in the future, we could return it properly to the UI, but we don't really have good support for returning values along with errors right now.